### PR TITLE
#317 Contact banner

### LINKED ISSUE #317: 🎨 Create Contact banner
#  Create Contact banner
---

#### Note: 
Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a user I want a contact page banner

### Description

As seen on the designs (link below), this section contains the following:

```
1. A title
2. Contact options
```

The contact section needs to appear as shown below (link to icons below)

Screenshot 2023-02-27 at 09 41 35

Your assignment is to implement the component to appear as above.

Please take note of all the design features.

#### Link to full design:
```
https://xd.adobe.com/view/7b72572b-b40a-4618-bfa6-5500ab28f13b-2e21/screen/a57c4c47-d3d7-48ce-9b3a-d224a4c624be
```

### Technical Analysis

This component will be created in  libs/pages/elewa/contact

The component will be called contact-banner

### Acceptance Criteria

1. When rendered the component is pixel perfect compared to the design above.
2. The component is clean and does not have more than 50 lines
3. The component SCSS uses CSS variables for colour, font, ... Connected to the main style.scss
4. The code is isolated to the libs/page/home package, unless a justified reason is specified.
5. The component is aligned using CSS flex
6. The design is responsive for screen, tablet and phone
7. The image renders fast and has multiple srcsets

### PR DESCRIPTION
**Pull request by timothymureithi**
# Description

This is a contact form banner for the contact page. Based on the requirements it is a responsive form that adjusts its layout based on the screen size of the device being used. The banner consists of a centered title, "Contacts", followed by three input fields: "Questions", "Press", and "Phone".

Example
This pull request is part of the work to make it easier for people to contribute to naming convention guides. 
One of the easiest way to make small changes would be using the Edit on Github button.

To achieve this, we needed to:

```
- Import contact-banner to contact-page
- Create a title for the banner and also create 3 input fields
- Write CSS code to ensure that the component is responsive on all devices
```

Fixes #317 - *Example*: Fixes # ([#317](https://github.com/italanta/elewa-group/issues/317))

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Screenshot (optional)

Screenshot from 2023-03-04 09-24-51
Screenshot from 2023-03-04 09-33-42

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 
Please also list any relevant details for your test configuration if necessary.
- [x] Test A
- [x] Test B

**Test Configuration**: (optional)
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by timothymureithi** [comment]
Created a new pull request under a new branch. I will close the other one after feedback.

File: libs/pages/elewa/contact/src/lib/components/contact-banner/contact-banner.component.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<div class="contact-banner">
+  <h1>Contact</h1>
+  <form class="contact-form" action="" method="post">
+    <div class="input-group">
+      <label for="questions">Questions</label>
+      <input type="text" id="questions" name="questions" placeholder="Enter your questions...">
+    </div>
+    <div class="input-group">
+      <label for="press">Press</label>
```
[the remark this answers is not in the document]
Comment (timothymureithi) [reply libs/pages/elewa/contact/src/lib/components/contact-banner/contact-banner.component.html:9]: I'm fixing up the code now

[the remark this answers is not in the document]
Comment (timothymureithi) [reply libs/pages/elewa/contact/src/lib/components/contact-banner/contact-banner.component.html]: Updated the changes. Please review

**Comment by IanOdhiambo9** [comment]
On mobile the width of the inputs is not uniform.

**Review by IanOdhiambo9 — CHANGES_REQUESTED** [review]
On mobile the width of the inputs is not uniform.

**Comment by timothymureithi** [comment]
> On mobile the width of the inputs is not uniform.

Let me fix it now.

**Comment by timothymureithi** [comment]
Fixed. Please review

**Pull request merged**